dash/dash.py

```diff
@@ -461,11 +464,9 @@ def index(self, *args, **kwargs):  # pylint: disable=unused-argument
         config = self._generate_config_html()
         metas = self._generate_meta_html()
         title = getattr(self, 'title', 'Dash')
-        if self._favicon:
-            favicon = '<link rel="icon" type="image/x-icon" href="{}">'.format(
-                self.get_asset_url(self._favicon))
-        else:
-            favicon = ''
+        favicon = '<link rel="icon" type="image/x-icon" href="{}">'.format(
+            self.get_asset_url(self._favicon) if self._favicon
+            else '{}_favicon'.format(self.config.requests_pathname_prefix))
```

Comment:
I found this logic with the nested formats and conditional expression a little hard to parse initially. I'd have a preference for something more verbose but easier to read, like:

```python
if self._favicon:
    favicon_url = self.get_asset_url(self._favicon)
else:
    favicon_url = '{}_favicon'.format(self.config.requests_pathname_prefix)
favicon = '<link rel="icon" type="image/x-icon" href="{}">'.format(favicon_url)
```